### ISSUE TITLE
Qt Creator - different (uglier) mimetype icons with Numix-Circle than Numix

### ISSUE THREAD

**Issue by LogicStuff** [numixproject/numix-core#3299]
Why does Qt Creator with Numix-Circle (light and dark) theme use these inferior icons in its project/file explorer? Excuse me if I did not diagnose this problem thoroughly, but I couldn't even find where the icons from the first image were located.

**Qt Creator with Numix-Circle:**

![a](https://cloud.githubusercontent.com/assets/11589772/16979844/9ef9882c-4e63-11e6-90e0-acff48c3e0d5.png)

**Qt Creator with Numix; other applications (Nautilus, Nemo) with Numix or Numix-Circle:**

![b](https://cloud.githubusercontent.com/assets/11589772/16979850/a7a5ce68-4e63-11e6-83e2-a7e79bec6a65.png)

I'd like to use Numix-Circle and have the later icons shown in Qt Creator. Is this problem on your side?

**Comment by palob** [comment]
Whoops, that's curious. The first icons don't even exist. It's the text editor icon from Numix Circle with ``C++`` / ``h`` overlays from somewhere else.

So you do have both Numix and Numix Circle installed?

The first screenshot is with Numix Circle (effectively Numix Circle + Numix) enabled and the second one with only Numix in use?

**Comment by LogicStuff** [comment]
Yes, I do have both. I have also quite a few other icon themes. I've gone through all of them and they all used different icons.

The second image shows that Numix with Qt Creator shows the desired icons, and that *there's no difference between the listed Numix and Numix-Circle mimetype icons in other programs (file browsers)*. I've seen those icons only in Qt Creator with Numix-Circle, nowhere else.

It looks like overlays created by Qt Creator - it shows C++ on a `.tpp` (template implementation) file. But why? Did it fail to load the Numix ones?

I have reinstalled Numix and Numix-Circle, nothing changed.

**Comment by palob** [comment]
@LogicStuff Can you delete the icon ``text.svg`` from your Numix Circle installation and see if that helps?

Maybe this is related to #3312.

**Comment by LogicStuff** [comment]
It's a little bit better, at least the usual icon for plain text files is shown. But the problem still remains:
![a](https://cloud.githubusercontent.com/assets/11589772/17112180/455ed7ea-52a5-11e6-881b-4fb92138321d.png)

**Comment by LogicStuff** [comment]
@palob
It's a little bit better, at least the usual icon for plain text files is shown. But the problem still remains:
![a](https://cloud.githubusercontent.com/assets/11589772/17112180/455ed7ea-52a5-11e6-881b-4fb92138321d.png)

**Comment by palob** [comment]
Wow, I can't make any sense of that.

**Comment by bil-elmoussaoui** [comment]
@LogicStuff Can you give us more informations about file manager version, and if you're using the latest version of Numix+ Numix Circle (github one). Also your distro & Gtk version.
I will see if i can reporduce that later

**Comment by LogicStuff** [comment]
@bil-elmoussaoui First, note that this is an issue with Qt Creator and I used file managers just for comparison (both GNOME Nautilus 3.18/3.20.1 and Nemo 2.8.7 are displaying the icons I want).

My setup is Ubuntu 16.04 with Unity and GNOME 3.20 (to which I mistakenly switched, although not having any major problems with this combination). I've had this problem for a few months now, even with Ubuntu 14.04, and I was using GNOME 3.18 on 16.04 before with same results.

My Gtk versions are `2.24.30-2ubuntu1~ppa0` and `3.20.6-2ubuntu1~xenial2`.

**Comment by bilelmoussaoui** [comment]
@LogicStuff It's an issue with QtCreator itself. They use the default text mimetype and add strings/images on it. Which makes the whole thing ugly. That's why the first icons were picking the `text.svg` icon and using it. This should be reported upstream to use the correct mimetype or just shipping their own icons instead of doing this. 
The issue happens with all the themes

**Comment by Foggalong** [comment]
@bil-elmoussaoui Why does it not pick up `text.svg` when just using the base theme?

**Comment by bilelmoussaoui** [comment]
It's what QtCreator does. They use one icon, and add the mimetype on top of it. Instead of using the c++ mimetype icon, they write c++ on the text.svg icon. You can check that easily either by switching to an other theme or replacing the text.svg icon with something else: 

Here's an example where i switched the text.svg icon with a folder icon

![screenshot from 2017-12-27 18-51-51](https://user-images.githubusercontent.com/7660997/34388946-03ef7d92-eb37-11e7-998c-40b18811b73c.png)

**Comment by palob** [comment]
But in his first post @LogicStuff showed an unbugged screenshot.
Do we even have ``text.svg`` in base?

**Comment by bilelmoussaoui** [comment]
@palob We have text.svg in base. Which is a symlink to application-text in the mimetype folder. i tried applying the Numix theme, but it's still the text.svg icon that's being used so i'm getting the same result as in my latest comment.

**Comment by palob** [comment]
I don't think it is supposed to be in base (neither in Circle/Square).

**Comment by Foggalong** [comment]
So if the mimetype icon is in the applied theme (ignoring inherits) it uses that, else it takes whatever `text.svg` is and then applies a symbol on top of it?

**Comment by bilelmoussaoui** [comment]
@Foggalong Yep! I don't know if removing text.svg from Base will fix it

**Comment by Foggalong** [comment]
@bil-elmoussaoui Why from base? I would have assumed this was either a problem from `text.svg` being in Circle/Square (which I think it might have to be because of text editor names?) or otherwise an upstream problem of not following inherits in lookup.

**Comment by Foggalong** [comment]
I've done some brainstorming/research and this is what's happening:

1. QtCreator looks for the C++ mimetypes in Numix Circle which don't exist.
2. It then looks for the fallback icon in Circle, `text.svg`, onto which it overlays the symbols
3. If `text.svg` is not in Circle it will start looking for it in the inherits hierarchy, starting with the Numix. base theme

While 2 isn't great practice either, 3 is where the problems really start because it should be looking for the original mimetypes in the inherited themes rather than the `text.svg` fallback. This is also why it works fine when base is used on its own.

TL;DR: we've long since deleted `text.svg` from core because it shouldn't really be used here anyway. However for this problem to get fixed it will require action on QtCreators end.
